spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/partitioner/Murmur3PartitionerTokenRangeSplitter.scala

```diff
@@ -1,34 +1,21 @@
 package com.datastax.spark.connector.rdd.partitioner
 
-import com.datastax.spark.connector.rdd.partitioner.dht.{LongToken, TokenFactory, TokenRange}
+import com.datastax.spark.connector.rdd.partitioner.dht.LongToken
 
-/** Fast token range splitter assuming that data are spread out evenly in the whole range.
-  * @param dataSize estimate of the size of the data in the whole ring */
-class Murmur3PartitionerTokenRangeSplitter(dataSize: Long)
+/** Fast token range splitter assuming that data are spread out evenly in the whole range. */
+private[partitioner] class Murmur3PartitionerTokenRangeSplitter
   extends TokenRangeSplitter[Long, LongToken] {
 
-  private val tokenFactory =
-    TokenFactory.Murmur3TokenFactory
+  private type TokenRange = com.datastax.spark.connector.rdd.partitioner.dht.TokenRange[Long, LongToken]
 
-  private type TR = TokenRange[Long, LongToken]
+  override def split(tokenRange: TokenRange, splitCount: Int): Seq[TokenRange] = {
+    val rangeTokenCount = tokenRange.distance
+    val splitPointsCount = if (rangeTokenCount < splitCount) rangeTokenCount.toInt else splitCount
```

Comment:
I'm a little confused why we are building this special case, Are we worried about a token range having a size of less than splitCount? (I mean this looks like the right thing to do but i'm not sure we need to do it?) I guess without this we have the worry of getting a SubRange that's empty...